src/rulesets/Base/nondiff.jl

```diff
@@ -289,6 +289,7 @@ VERSION >= v"1.3" && @non_differentiable istaskfailed(::Task)
 VERSION >= v"1.4" && @non_differentiable only(::Char)
 @non_differentiable open(::Any)
 
+@non_differentiable partialsortperm(::AbstractArray{Any}, ::Union{Integer, OrdinalRange})
```

Comment:
```suggestion
@non_differentiable partialsortperm(::AbstractVector{Any}, ::Union{Integer, OrdinalRange})
```
I think this is how the `partialsortperm` is constrained, so we should probably do the same